### ISSUE TITLE
Dimensions.get("window").width returns bad value sometimes

### ISSUE THREAD

**Issue by adisoftbn** [Microsoft/react-native-windows#844]
Hello,

I'm trying to scale my application UI to make it fit perfect on every screen.

While using android and iOS i saw that **Dimensions.get("window").width** returns all the time the same value in **Device independent pixels**.

When I rebuild solution in Visual studio at first run i get **500 pixels** which is correct.
If I run it second time without rebuilding it i get my desktop screen resolution which is **1920 pixels** :) even if the app window has the same size **500 pixels**.

my code is :

```
import { Dimensions } from 'react-native';
console.log('window width: '+Dimensions.get("window").width);
```

**Comment by rozele** [comment]
There's an event that is fired that lets the app know when the dimensions change.  Are you tracking that event?  If you're running off the packager server, the second run uses a cached bundle, and the point at which the window is measured is an earlier stage when UWP has not determined the correct Window size yet.  It will update correctly once the root view mounts.

**Comment by adisoftbn** [comment]
Thank you @rozele for response. :D

I need to calculate it before i run any other scripts, because all styles are scaled using the value I get.
Is there any way to figure out if i'm on Windows Mobile or Windows Desktop ? I want to disable it on desktop.

On mobile works correctly. I have this problem only when I test on the desktop version of windows.

I have only few issues left. :D

Best regards, 
Adrian

**Comment by rozele** [comment]
Good point, I think we can extend the Platform module to include the device family

**Comment by ndbroadbent** [comment]
I'm experiencing a similar issue.

When the UWP desktop app first starts, `Dimensions.get('window').width` returns `1280`.  This causes some of my layout to be incorrect. 

If I carefully resize the window by a couple of pixels, then `Dimensions.get('window').width` jumps to `900`, which produces the correct layout.

I think the width is not being calculated correctly when the app first starts, but it is corrected whenever I resize the window, or reload JS in development.

**Comment by ndbroadbent** [comment]
Here's some more info. I should mention that I'm on the branch `0.42-stable`.

I [added a breakpoint in GetDimensions](https://github.com/Microsoft/react-native-windows/blob/master/ReactWindows/ReactNative/UIManager/UIManagerModule.cs#L558).

On the first call I get:

```
		bounds	{0,0,1280,800}	Windows.Foundation.Rect
		scale	1.5	double
```

As soon as I resize the window by 1 pixel, I get:

```
		bounds	{202,108,903.3333,588}	Windows.Foundation.Rect
		scale	1.5	double
```

**Comment by ndbroadbent** [comment]
AH! @adisoftbn - now I know what you mean about the first run being correct. This is because on the first run, we're loading the JS from the packager, which changes the time that Dimensions are being calculated. When you run the app again, apparently it uses some cached JS (even in development), so the bug appears. The bug also happens when you run a build with bundled JS.

I've just figured out one way to fix this, by recalculating Dimensions after adding the root view. I sent a PR: #1124

**Comment by ndbroadbent** [comment]
Ah, @rozele I just re-read you comments and noticed:

>  It will update correctly once the root view mounts.

That was not happening, which my PR fixes.

**Comment by rozele** [comment]
Seems like @ndbroadbent fixed this with #1124
